tap_mongodb/__init__.py

```diff
@@ -250,26 +251,44 @@ def main_impl():
     """
     args = utils.parse_args(REQUIRED_CONFIG_KEYS)
     config = args.config
+    srv = config.get('srv') is True
+
+    if not srv:
+        args = utils.parse_args(REQUIRED_CONFIG_KEYS_NON_SRV)
+        config = args.config
 
     # Default SSL verify mode to true, give option to disable
     verify_mode = config.get('verify_mode', 'true') == 'true'
     use_ssl = config.get('ssl') == 'true'
 
-    connection_params = {"host": config['host'],
-                         "port": int(config['port']),
-                         "username": config.get('user', None),
-                         "password": config.get('password', None),
-                         "authSource": config['auth_database'],
-                         "ssl": use_ssl,
-                         "replicaSet": config.get('replica_set', None),
-                         "readPreference": 'secondaryPreferred'
-                         }
-
-    # NB: "ssl_cert_reqs" must ONLY be supplied if `SSL` is true.
+    connection_query = {
+        'readPreference': 'secondaryPreferred',
+        'authSource': config['auth_database'],
+    }
+
+    if config.get('replica_set'):
+        connection_query['replicaSet'] = config['replica_set']
+
+    if use_ssl:
+        connection_query['ssl'] = 'true'
+
+    # NB: "sslAllowInvalidCertificates" must ONLY be supplied if `SSL` is true.
     if not verify_mode and use_ssl:
-        connection_params["ssl_cert_reqs"] = ssl.CERT_NONE
+        connection_query['sslAllowInvalidCertificates'] = 'true'
```

Comment:
This should be `tlsAllowInvalidCertificates`

Reply:
@ivanovyordan this one too

Reply:
Done